src/core/thread/mle.cpp

```diff
@@ -3874,7 +3946,270 @@ void Mle::HandleLinkProbe(const Message &aMessage, const Ip6::MessageInfo &aMess
+    // Find Child
+    aMessageInfo.GetPeerAddr().GetIid().ConvertToExtAddress(extAddr);
+
+    peerSsed = Get<ChildTable>().FindChild(extAddr, Child::kInStateAnyExceptInvalid);
+
+    if (peerSsed == nullptr)
+    {
+        VerifyOrExit((peerSsed = Get<ChildTable>().GetNewChild()) != nullptr, error = OT_ERROR_NO_BUFS);
+
+        // MAC Address
+        peerSsed->SetExtAddress(extAddr);
+        peerSsed->GetLinkInfo().Clear();
+        peerSsed->GetLinkInfo().AddRss(aMessageInfo.GetThreadLinkInfo()->GetRss());
+        peerSsed->ResetLinkFailures();
+        peerSsed->SetState(Neighbor::kStateLinkRequest);
+        peerSsed->SetDeviceMode(DeviceMode(DeviceMode::kModeRxOnWhenIdle));
+        peerSsed->SetRloc16(sourceAddress);
+    }
+
+    // Send Response
+    SuccessOrExit(error = SendLinkAccept(aMessageInfo, peerSsed, challenge, true));
+
+exit:
+    return error;
+}
+
+otError Mle::HandleLinkAccept(const Message &aMessage, const Ip6::MessageInfo &aMessageInfo, bool aRequest)
+{
+    otError         error    = OT_ERROR_NONE;
+    Child *         peerSsed = nullptr;
+    Mac::ExtAddress extAddr;
+    uint16_t        sourceAddress;
+    uint16_t        version;
+    uint8_t         mode;
+    Challenge       response;
+
+    // Source Address
+    SuccessOrExit(error = Tlv::Find<SourceAddressTlv>(aMessage, sourceAddress));
+
+    // Ignore router's Link Request
+    VerifyOrExit(!IsActiveRouter(sourceAddress), error = OT_ERROR_DROP);
+
+    // Find Child
+    aMessageInfo.GetPeerAddr().GetIid().ConvertToExtAddress(extAddr);
+    peerSsed = Get<ChildTable>().FindChild(extAddr, Child::kInStateAnyExceptInvalid);
+
+    VerifyOrExit(peerSsed != nullptr, error = OT_ERROR_DROP);
+
+    // Mode
+    SuccessOrExit(error = Tlv::Find<ModeTlv>(aMessage, mode));
+    VerifyOrExit(!(mode & (DeviceMode::kModeFullThreadDevice | DeviceMode::kModeRxOnWhenIdle)), error = OT_ERROR_DROP);
+
+    // Verify response
+    SuccessOrExit(error = ReadResponse(aMessage, response));
+    VerifyOrExit(response.Matches(peerSsed->GetChallenge(), peerSsed->GetChallengeSize()), error = OT_ERROR_SECURITY);
+
+    // Version
+    SuccessOrExit(error = Tlv::Find<VersionTlv>(aMessage, version));
+    VerifyOrExit(version >= OT_THREAD_VERSION_1_2, error = OT_ERROR_PARSE);
+
+    // Finish link synchronization
+    peerSsed->SetExtAddress(extAddr);
+    peerSsed->SetVersion(version);
+    peerSsed->SetRloc16(sourceAddress);
+    peerSsed->SetLastHeard(TimerMilli::GetNow());
+    peerSsed->SetDeviceMode(DeviceMode(mode));
+    peerSsed->GetLinkInfo().Clear();
+    peerSsed->GetLinkInfo().AddRss(aMessageInfo.GetThreadLinkInfo()->GetRss());
+    peerSsed->ResetLinkFailures();
+    peerSsed->SetState(Neighbor::kStateValid);
+
+    // Send Link Accept
+    if (aRequest)
+    {
+        Challenge challenge;
+
+        // Challenge
+        SuccessOrExit(error = ReadChallenge(aMessage, challenge));
+        SuccessOrExit(error = SendLinkAccept(aMessageInfo, peerSsed, challenge, false));
```

Comment:
As mentioned above, we can add default paramenter value and call this method without a 4th parameter.